USAGE.md

```diff
@@ -30,6 +30,13 @@ The recursive flag `-r` or `--recursive` will tell the scanner to search all sub
 
 Git directories are searched for the latest commit hash. Searching for git commit hash is intended to work with projects that use git submodules or a similar mechanism where dependencies are checked out as real git repositories. 
 
+### Ignored files
+
+By default, OSV-Scanner will not scan files that are ignored by `.gitignore` files. If the specified file is not part of a git repository, `.gitignore` files are parsed recursively starting from the target directory,
```

Comment:
"recursively" seems a bit vague -- can you clarify what this means? 

That said -- does the behaviour actually make sense? If a file isn't part of a git repo, no gitignores should apply to it right?

Reply:
CC @another-rex

Reply:
"recursively" meaning it parses `foo/.gitignore`, `foo/bar/.gitignore`, `foo/bar/baz/.gitignore`, etc.
I guess it'd be more clear to say "`.gitignore` files are parsed in the target directory and all subdirectories"?

I think this behaviour is the probably the most useful:
- I don't think you'd typically have `.gitignore` files in directories that aren't git repositories.
   - If you do, I would assume that you'd want to use it.
- Say you have a folder `foo/`, that's not a repo, and inside have `foo/bar/` and `foo/baz/` that are git repos. Running `osv-scanner -r foo/` should pick up the gitignore files from `bar/` and `baz/`.

Reply:
The way this is worded was rather unclear. It also doesn't clearly distinguish "specified file" from "target directory", and it wasn't clear what "not part of a git repository" means exactly. 

Would this alternative wording work:  

"All recursively scanned files are matched to a git repository (if it exists) and any corresponding .gitignore files are taken into account."

So just to make sure my own understanding is correct, using your example:  

Given `foo/` (not a git repo), `foo/bar` (git repo), and `foo/baz` (git repo):
- Scanning `foo/bar/blah` would use `foo/bar/.gitignore` if it exists
- Scanning `foo/baz/blah` would use `foo/baz/.gitignore` if it exists
- Scanning `foo/blah` would **also** use `foo/.gitignore` if it exists, even though it's not a git repo? 

There are variations of edge cases here. e.g. 
- If `foo/` is not a git repo AND has a gitignore,, and `foo/bar` is a git repo does not have a git ignore, does scannign something in `foo/bar` use the gitignore from `foo/` ?

Reply:
> So just to make sure my own understanding is correct, using your example:

These are all correct, as it currently works.

> Would this alternative wording work:
>
>"All recursively scanned files are matched to a git repository (if it exists) and any matching .gitignore files within that repository are taken into account."

I suppose, though I don't think it captures the (current) non-repository behaviour (e.g. dot point 3)

Reply:
> > So just to make sure my own understanding is correct, using your example:
> 
> These are all correct, as it currently works.
> 
> > Would this alternative wording work:
> > "All recursively scanned files are matched to a git repository (if it exists) and any matching .gitignore files within that repository are taken into account."
> 
> I suppose, though I don't think it captures the (current) non-repository behaviour (e.g. dot point 3)

There's a bunch of edge cases (e.g. the last one I listed in my last comment) that would be resolved if we more closely followed git behaviour (which should take into account repo boundaries). Is that easy to support?

Reply:
Tracking repo boundaries correctly is a bit tricky. I've made the bug #209 for this.